src/Kestrel.Core/KestrelConfigurationLoader.cs

```diff
@@ -0,0 +1,297 @@
+        /// for this type of endpoint.
+        /// </summary>
+        public KestrelConfigurationLoader LocalhostEndpoint(int port) => LocalhostEndpoint(port, options => { });
+
+        /// <summary>
+        /// Listens on ::1 and 127.0.0.1 with the given port. Requesting a dynamic port by specifying 0 is not supported
+        /// for this type of endpoint.
+        /// </summary>
+        public KestrelConfigurationLoader LocalhostEndpoint(int port, Action<ListenOptions> configure)
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            EndpointsToAdd.Add(() =>
+            {
+                Options.ListenLocalhost(port, configure);
+            });
+
+            return this;
+        }
+
+        /// <summary>
+        /// Listens on all IPs using IPv6 [::], or IPv4 0.0.0.0 if IPv6 is not supported.
+        /// </summary>
+        public KestrelConfigurationLoader AnyIPEndpoint(int port) => AnyIPEndpoint(port, options => { });
+
+        /// <summary>
+        /// Listens on all IPs using IPv6 [::], or IPv4 0.0.0.0 if IPv6 is not supported.
+        /// </summary>
+        public KestrelConfigurationLoader AnyIPEndpoint(int port, Action<ListenOptions> configure)
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            EndpointsToAdd.Add(() =>
+            {
+                Options.ListenAnyIP(port, configure);
+            });
+
+            return this;
+        }
+
+        /// <summary>
+        /// Bind to given Unix domain socket path.
+        /// </summary>
+        public KestrelConfigurationLoader UnixSocketEndpoint(string socketPath) => UnixSocketEndpoint(socketPath, _ => { });
+
+        /// <summary>
+        /// Bind to given Unix domain socket path.
+        /// </summary>
+        public KestrelConfigurationLoader UnixSocketEndpoint(string socketPath, Action<ListenOptions> configure)
+        {
+            if (socketPath == null)
+            {
+                throw new ArgumentNullException(nameof(socketPath));
+            }
+            if (socketPath.Length == 0 || socketPath[0] != '/')
+            {
+                throw new ArgumentException(CoreStrings.UnixSocketPathMustBeAbsolute, nameof(socketPath));
+            }
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            EndpointsToAdd.Add(() =>
+            {
+                Options.ListenUnixSocket(socketPath, configure);
+            });
+
+            return this;
+        }
+
+        /// <summary>
+        /// Open a socket file descriptor.
+        /// </summary>
+        public KestrelConfigurationLoader HandleEndpoint(ulong handle) => HandleEndpoint(handle, _ => { });
```

Comment:
Just curious - did we expose `ulong`s in the public API before?

Reply:
Yes these are all mirrors of the existing Listen APIs.

Reply:
Yes. `KestrelServerOptions.ListenHandle(ulong)`.

Reply:
:+1: